src/main.cpp

```diff
@@ -31,7 +32,15 @@ int main(int argc, char* argv[])
   switch (settings::run_mode) {
   case RunMode::FIXED_SOURCE:
   case RunMode::EIGENVALUE:
-    err = openmc_run();
+    switch (settings::solver_type) {
+    case SolverType::MONTE_CARLO:
+      err = openmc_run();
+      break;
+    case SolverType::RANDOM_RAY:
+      openmc_run_random_ray();
+      err = 0;
```

Comment:
Why not have `err = openmc_run_random_ray()`?

Reply:
That function doesn't return anything, and error handling is not done in that manner in the random ray solver. It actually did originally just return 0 -- see above discussion from another reviewer on this topic.